cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp

```diff
@@ -458,11 +458,7 @@ void RenderAdditiveLightQueue::updateCameraUBO(const Camera *camera, gfx::Comman
     memcpy(uboCameraView.data() + UBOCamera::MAT_VIEW_PROJ_INV_OFFSET, camera->matViewProjInv.m, sizeof(cc::Mat4));
     TO_VEC3(uboCameraView, camera->position, UBOCamera::CAMERA_POS_OFFSET);
 
-    auto projectionSignY = device->getScreenSpaceSignY();
-    if (camera->getWindow()->hasOffScreenAttachments) {
-        projectionSignY *= device->getUVSpaceSignY(); // need flipping if drawing on render targets
-    }
-    uboCameraView[UBOCamera::CAMERA_POS_OFFSET + 3] = projectionSignY;
+    uboCameraView[UBOCamera::CAMERA_POS_OFFSET + 3] = (uint)(device->getScreenSpaceSignY() * 0.5 + 0.5) << 1 | (uint)(device->getClipSpaceSignY() * 0.5 + 0.5);
```

Comment:
Can add a function to do `device->getScreenSpaceSignY() * 0.5 + 0.5) << 1 | (uint)(device->getClipSpaceSignY() * 0.5 + 0.5)`?

Reply:
And please do not use c style cast.

Reply:
> Can add a function to do `device->getScreenSpaceSignY() * 0.5 + 0.5) << 1 | (uint)(device->getClipSpaceSignY() * 0.5 + 0.5)`?

Yes, add it to the device?

Reply:
> And please do not use c style cast.

Roger that

Reply:
Declare one inside the pipeline will do. Never attach irrelevant custom logic to the lower layer.

Reply:
> Declare one inside the pipeline will do. Never attach irrelevant custom logic to the lower layer.

got it